src/sentry/demo/tasks.py

```diff
@@ -18,19 +24,49 @@ def delete_users_orgs(**kwargs):
     # delete everything older than a day
     cutoff_time = timezone.now() - timedelta(days=1)
 
-    # first mark orgs for deletion
     # note this only runs in demo mode (not SaaS) so the underlying tables here are small
     org_list = Organization.objects.filter(
-        date_added__lte=cutoff_time, flags=F("flags").bitor(Organization.flags["demo_mode"])
+        demoorganization__isnull=False,
```

Comment:
You can actually skip this check, since you're filtering on other properties of `demoorganization`